File: example/admin_request.ts
import jwt from "jsonwebtoken"; // npm install jsonwebtoken
import fs from "fs";
import { v4 as uuidv4 } from "uuid"; // npm install uuid

const AUD = "shortener";
const ISS = "dev@guigiusti.com";
const USER = "guigiusti";
const SUB = "dev@guigiusti.com";
const TOKEN_EXPIRATION = 60;

const HOST = "http://localhost";

const privateKey = fs.readFileSync("../private.pem", "utf8");

function getToken() {
    const now = Math.floor(Date.now() / 1000);
    const jwtPayload = {
        aud: AUD,
        iss: ISS,
        iat: now,
        nbf: now,
        exp: now + TOKEN_EXPIRATION,
        sub: SUB,
        jti: uuidv4(),
        user: USER
    };
    return jwt.sign(jwtPayload, privateKey, { algorithm: "ES256" });
}

async function getUrls() {
    const token = getToken();
    const response = await fetch(`${HOST}/admin/urls`, {
        headers: { "Authorization": `Bearer ${token}` }
    });
    return response.json();
}

async function createUrl(url: string, shortId: string) {
    const token = getToken();
    const postResponse = await fetch(`${HOST}/admin/urls`, {
        method: "POST",
        headers: { "Authorization": `Bearer ${token}` },
        body: JSON.stringify({
            url: url,
            shortId: shortId
        })
    });
    return postResponse.json();
}

async function updateUrl(url: string, shortId: string) {
    const token = getToken();
    const putResponse = await fetch(`${HOST}/admin/urls`, {
        method: "PUT",
        headers: { "Authorization": `Bearer ${token}` },
        body: JSON.stringify({
            url: url,
            shortId: shortId
        })
    });
    return putResponse.json();
}

async function deleteUrl(shortId: string) {
    const token = getToken();
    const deleteResponse = await fetch(`${HOST}/admin/urls`, {
        method: "DELETE",
        headers: { "Authorization": `Bearer ${token}` },
        body: JSON.stringify({ shortId: shortId })
    });
    return deleteResponse.json();
}

//createUrl("https://guigiusti.com", "guigiusti");
// updateUrl("https://guigiusti.com/blog", "guigiusti");
// deleteUrl("guigiusti");
// console.log(getUrls());

File: example/admin_request.py
import jwt # pip install pyjwt
import requests
from pathlib import Path
import time
import uuid

AUD="shortener"
ISS="dev@guigiusti.com"
USER="guigiusti"
SUB="dev@guigiusti.com"
TOKEN_EXPIRATION=60

HOST="http://localhost"

private_key = Path("../private.pem").read_bytes()

def get_token():
    now = int(time.time())
    jwt_payload = {
        "aud": AUD,
        "iss": ISS,
        "iat": now,
        "nbf": now,
        "exp": now + TOKEN_EXPIRATION,
        "sub": SUB,
        "jti": str(uuid.uuid4()),
        "user": USER
    }
    return jwt.encode(jwt_payload, private_key, algorithm="EdDSA")

def get_urls():
    token = get_token()
    response = requests.get(
        f"{HOST}/admin/urls",
        headers={"Authorization": f"Bearer {token}"}
    )
    return response.json()

def create_url(url, shortId):
    token = get_token()
    post_response = requests.post(
    f"{HOST}/admin/urls",
    headers={"Authorization": f"Bearer {token}"},
    body={
        "url": url,
        "shortId": shortId
    }
)
    return post_response.json()
def update_url(url, shortId):
    token = get_token()
    put_response = requests.put(
        f"{HOST}/admin/urls",
        headers={"Authorization": f"Bearer {token}"},
        body={
            "url": url,
            "shortId": shortId
        }
    )
    return put_response.json()

def delete_url(shortId):
    token = get_token()
    delete_response = requests.delete(
        f"{HOST}/admin/urls",
        headers={"Authorization": f"Bearer {token}"},
        body={"shortId": shortId}
    )
    return delete_response.json()

# create_url("https://guigiusti.com", "guigiusti")
# update_url("https://guigiusti.com/blog", "guigiusti")
# delete_url("guigiusti")
#print(get_urls())